gpMgmt/bin/analyzedb

```diff
@@ -856,7 +856,7 @@ class AnalyzeDb(Operation):
         2. The leaf partitions (if range partitioned, especially by date) will be ordered in descending
            order of the partition key, so that newer partitions can be analyzed first.
         """
-        candidate_regclass_str = get_oid_str(candidates + list(root_partition_col_dict.keys()))
+        candidate_regclass_str = get_oid_str(candidates.union(set(root_partition_col_dict.keys())))
```

Comment:
Looks like it is not necessary to push set into get_oid_str. itertools [chain](https://docs.python.org/2/library/itertools.html#itertools.chain) function can help avoid candidates copying in memory.